store/src/main/java/org/apache/rocketmq/store/CommitLog.java

```diff
@@ -1154,7 +1249,7 @@ public AppendMessageResult doAppend(final long fileFromOffset, final ByteBuffer
 
             if (propertiesLength > Short.MAX_VALUE) {
                 log.warn("putMessage message properties length too long. length={}", propertiesData.length);
-                return new AppendMessageResult(AppendMessageStatus.PROPERTIES_SIZE_EXCEEDED);
+                throw new RuntimeException("PROPERTIES_SIZE_EXCEEDED");
```

Comment:
By changing to this, `PROPERTIES_SIZE_EXCEEDED` won't be caught in the following switch.